packages/internal/bridge/sdk/src/tokenBridge.ts

```diff
@@ -302,82 +296,18 @@ export class TokenBridge {
   public async getUnsignedApproveBridgeTx(
     req: ApproveBridgeRequest,
   ): Promise<ApproveBridgeResponse> {
-    await this.validateChainConfiguration();
-    return await this.getUnsignedApproveBridgeTxPrivate(req);
-  }
-
-  public async getUnsignedApproveBridgeTxPrivate(
-    req: ApproveBridgeRequest,
-  ): Promise<ApproveBridgeResponse> {
-    await this.validateDepositArgs(
-      req.token,
-      req.senderAddress,
-      req.amount,
-      req.sourceChainId,
-      req.destinationChainId,
-    );
-
-    // If the token is NATIVE, no approval is required
-    if (req.token.toUpperCase() === 'NATIVE') {
-      return {
-        contractToApprove: null,
-        unsignedTx: null,
-      };
-    }
-
-    let sourceProvider:ethers.providers.Provider;
-    let sourceBridgeAddress: string;
-    if (req.sourceChainId === this.config.bridgeInstance.rootChainID) {
-      sourceProvider = this.config.rootProvider;
-      sourceBridgeAddress = this.config.bridgeContracts.rootERC20BridgeFlowRate;
-    } else {
-      sourceProvider = this.config.childProvider;
-      sourceBridgeAddress = this.config.bridgeContracts.childERC20Bridge;
-    }
-
-    const erc20Contract: ethers.Contract = await withBridgeError<ethers.Contract>(
-      async () => new ethers.Contract(req.token, ERC20, sourceProvider),
-      BridgeErrorType.PROVIDER_ERROR,
-    );
-
-    // Get the current approved allowance of the RootERC20Predicate
-    const allowance: ethers.BigNumber = await withBridgeError<ethers.BigNumber>(() => erc20Contract
-      .allowance(
-        req.senderAddress,
-        sourceBridgeAddress,
-      ), BridgeErrorType.PROVIDER_ERROR);
-
-    // If the allowance is greater than or equal to the deposit amount, no approval is required
-    if (allowance.gte(req.amount)) {
-      return {
-        contractToApprove: null,
-        unsignedTx: null,
-      };
-    }
-    // Calculate the amount of tokens that need to be approved for deposit
-    const approvalAmountRequired = req.amount.sub(
-      allowance,
-    );
-
-    // Encode the approve function call data for the ERC20 contract
-    const data: string = await withBridgeError<string>(async () => erc20Contract.interface
-      .encodeFunctionData('approve', [
-        sourceBridgeAddress,
-        approvalAmountRequired,
-      ]), BridgeErrorType.INTERNAL_ERROR);
-
-    // Create the unsigned transaction for the approval
-    const unsignedTx: ethers.providers.TransactionRequest = {
-      data,
-      to: req.token,
-      value: 0,
-      from: req.senderAddress,
-      chainId: parseInt(req.sourceChainId, 10),
-    };
-
+    const res = await this.getUnsignedBridgeBundledTx({
```

Comment:
is my understanding correct that calling `getUnsignedBridgeBundledTx()` here would perform a number of subsequent calls (through `getUnsignedBridgeBundledTxPrivate()`) that are not related to getting the the approval transaction? I imagine these calls are probably expensive?

Reply:
Yes, that's right. I was thinking about deprecating these two function calls `getUnsignedApproveBridgeTx` and `getUnsignedBridgeTx` or maybe there is a use case for each call individually? 🤔  WDYT?